src/utils/function_metadata.rs

```diff
@@ -8,17 +8,22 @@ pub struct FunctionMetadata {
     id: usize,
     variant: usize,
     returns: Type,
+    is_all_caps: bool,
 }
 
 impl FunctionMetadata {
-    pub fn new<T: Into<String>>(name: T, id: usize, variant: usize, returns: &Type) -> Self {
+    pub fn new<T: Into<String>>(name: T, id: usize, variant: usize, returns: &Type, is_all_caps: bool) -> Self {
         let name = name.into();
         let returns = returns.clone();
-        FunctionMetadata { name, id, variant, returns }
+        FunctionMetadata { name, id, variant, returns, is_all_caps }
```

Comment:
This `is_all_caps` should be calculated on the go instead of storing it as a variable.